llama_stack/providers/remote/vector_io/pgvector/pgvector.py

```diff
@@ -168,6 +310,43 @@ async def delete_chunk(self, chunk_id: str) -> None:
         with self.conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cur:
             cur.execute(f"DELETE FROM {self.table_name} WHERE id = %s", (chunk_id,))
 
+    def get_distance_metric(self) -> str:
+        """
+        Returns: the distance metric used for vector search.
+        """
+        return self.distance_metric
+
+    def get_pgvector_search_operator(self) -> str:
+        return self.PGVECTOR_DISTANCE_METRIC_TO_SEARCH_OPERATOR[self.distance_metric]
+
+    def check_distance_metric_availability(self, distance_metric: str) -> None:
+        """Check if the distance metric is supported by PGVector.
+
+        Args:
+            distance_metric: The distance metric to check
+
+        Raises:
+            ValueError: If the distance metric is not supported
+        """
+        if distance_metric not in self.PGVECTOR_DISTANCE_METRIC_TO_SEARCH_OPERATOR:
+            supported_metrics = list(self.PGVECTOR_DISTANCE_METRIC_TO_SEARCH_OPERATOR.keys())
+            raise ValueError(
+                f"Distance metric '{distance_metric}' is not supported by PGVector. "
+                f"Supported metrics are: {', '.join(supported_metrics)}"
+            )
+
+    def set_distance_metric(self, distance_metric: str) -> None:
```

Comment:
`query_vector` is a part of same class where we are providing distance_metric. Can we avoid using a setter here, to simply the code?